rpc/src/v1/impls/eth.rs

```diff
@@ -672,18 +672,22 @@ impl<C, M> EthFilter for EthFilterClient<C, M> where
 							to_value(&hashes)
 						},
 						PollFilter::PendingTransaction(ref mut previous_hashes) => {
+							// get hashes of pending transactions
 							let current_hashes = take_weak!(self.miner).pending_transactions_hashes();
-							// calculate diff
 							let previous_hashes_set = previous_hashes.into_iter().map(|h| h.clone()).collect::<HashSet<H256>>();
```

Comment:
Try calling `previous_hashes.iter()` to explicitly get an iterator over `&H256` rather than deferring to the `IntoIterator` implementation for `&mut Vec<H256>` which has item `&mut H256`. Then I think cloned should work.